app/main.py

```diff
@@ -1,23 +1,27 @@
+import asyncio
 import time
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def main() -> None:
     # create an IOT service
     service = IOTService()
 
     # create and register a few devices
     hue_light = HueLightDevice()
     speaker = SmartSpeakerDevice()
     toilet = SmartToiletDevice()
-    hue_light_id = service.register_device(hue_light)
-    speaker_id = service.register_device(speaker)
-    toilet_id = service.register_device(toilet)
+    hue_light_id = await service.register_device(hue_light)
+    speaker_id = await service.register_device(speaker)
+    toilet_id = await service.register_device(toilet)
+
+    async def coroutines(*coroutines) -> None:
+        tasks = [asyncio.create_task(coro) for coro in coroutines]
+        await asyncio.gather(*tasks)
```

Comment:
Check the requirements.
You have to change the wake_up_program and sleep_program blocks and their run — to the combination of the run_sequence and run_parallel sending messages to the service to make it more logical. Please do that separately, firstly for wake_up, then for sleep.